backend/src/services/auto_trader.py

```
@@ -301,23 +322,31 @@ async def _analyze_and_buy(self, token: Dict):
+                    score = rug_report.get('score', 0)
+                    max_allowed_score = self.config.get("rugcheck_max_score", 5000)
+
+                    if score > max_allowed_score:
+                        logger.warning(f"AutoTrader: Skipping {token.get('symbol', token_address)} - RugCheck score ({score}) exceeds max allowed ({max_allowed_score}).")
+                        return
+                    logger.info(f"AutoTrader: RugCheck passed for {token_address} with score: {score}")
+                    break # Success
+                else:
+                    logger.warning(f"AutoTrader: RugCheck API returned status {response.status_code}. Proceeding with caution.")
+                    break # Don't retry on non-200 if it's not a timeout/connection error
+            except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)